website/public/js/controllers/rootCtrl.js

```diff
@@ -36,6 +36,8 @@ habitrpg.controller("RootCtrl", ['$scope', '$rootScope', '$location', 'User', '$
     $rootScope.toJson = angular.toJson;
     $rootScope.Payments = Payments;
 
+    $http.defaults.headers.common['x-client'] = "habitica-web";
```

Comment:
it's set twice, because rootCtrl isn't loaded yet before registration/login, but the header was needed during registration.